pkg/framework/host.go

```diff
@@ -316,15 +316,41 @@ func (h *Host) InstallOperator(name, cr, values, version string) error {
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	podName, err := h.PodName(h.targetNamespace, fmt.Sprintf("app=%s", name))
-	if err != nil {
+	// TODO introduced: https://github.com/giantswarm/e2e-harness/pull/121
+	// This fallback from h.targetNamespace was introduced because not all our
+	// operators accept and apply configured namespaces.
+	//
+	// Tracking issue: https://github.com/giantswarm/giantswarm/issues/4123
+	//
+	// Final version of the code:
+	//
+	//	podName, err := h.PodName(h.targetNamespace, fmt.Sprintf("app=%s", name))
+	//	if err != nil {
+	//		return microerror.Mask(err)
+	//	}
+	//	err = h.filelogger.StartLoggingPod(h.targetNamespace, podName)
+	//	if err != nil {
+	//		return microerror.Mask(err)
+	//	}
+	//
+	podNamespace := h.targetNamespace
+
+	podName, err := h.PodName(podNamespace, fmt.Sprintf("app=%s", name))
+	if IsNotFound(err) {
+		podNamespace = "giantswarm"
+		podName, err = h.PodName(podNamespace, fmt.Sprintf("app=%s", name))
```

Comment:
To me it looks like we can directly use the target namespace everywhere but here and simply use the hardcoded alternative. The final version of the code would then be to remove the not found exception with its work around code of 4 lines.

Reply:
We unfortunately also need it when we start the logging on the pod: https://github.com/giantswarm/e2e-harness/blob/ec5dbb5fb89a95b4b1d922ff13ee481c8cccdf2e/pkg/framework/host.go#L349